tar.gz package fails to install without npm/gulp

Running `pip install` on the .tar.gz version of the package fails on a system without a global installation of npm / gulp - this is because the installation process calls the `python setup.py egg_info` command, which triggers the npm build.

    $ pip install https://pypi.python.org/packages/b8/45/bc2cacc434c579d8dfb80a829d330ca83b97269efc12cbccd80832b86b3a/wagtailfontawesome-1.1.1.tar.gz#md5=a396d0775ca8eae185fd83d43afdf8c7
    Collecting https://pypi.python.org/packages/b8/45/bc2cacc434c579d8dfb80a829d330ca83b97269efc12cbccd80832b86b3a/wagtailfontawesome-1.1.1.tar.gz#md5=a396d0775ca8eae185fd83d43afdf8c7
      Using cached wagtailfontawesome-1.1.1.tar.gz
        Complete output from command python setup.py egg_info:
        running egg_info
        npm ERR! path /private/var/folders/p6/_51b92q88xj9kk0059bc8f3h0000gn/T/pip-y7DmTG-build/package.json
        npm ERR! code ENOENT
        npm ERR! errno -2
        npm ERR! syscall open
        npm ERR! enoent ENOENT: no such file or directory, open '/private/var/folders/p6/_51b92q88xj9kk0059bc8f3h0000gn/T/pip-y7DmTG-build/package.json'
        npm ERR! enoent This is related to npm not being able to find a file.
        npm ERR! enoent

        npm ERR! A complete log of this run can be found in:
        npm ERR!     /Users/matthew/.npm/_logs/2017-11-27T15_01_58_649Z-debug.log
        Error compiling assets: Command '[u'npm', u'run', u'build']' returned non-zero exit status 254

        ----------------------------------------
    Command "python setup.py egg_info" failed with error code 1 in /private/var/folders/p6/_51b92q88xj9kk0059bc8f3h0000gn/T/pip-y7DmTG-build/

(Related: when I try to build the package myself, it's missing README.md since that's not listed in MANIFEST.in, and so setup fails at the point where it tries to build `long_description` from the readme. However, README.md _is_ present in the current release .tar.gz, so I guess your build process is slightly different from mine...)

--- a/setup.py
+++ b/setup.py
@@ -7,7 +7,8 @@ import subprocess
 from os import path
 
 from setuptools import find_packages, setup
-from setuptools.command.egg_info import egg_info as base_egg_info
+from setuptools.command.sdist import sdist as base_sdist
+from setuptools.command.bdist_egg import bdist_egg as base_bdist_egg
 
 from wagtailfontawesome import __version__
 
@@ -66,17 +67,25 @@ def md2pypi(filename):
 long_description = md2pypi('README.md')
 
 
-class egg_info(base_egg_info):
-    def run(self):
-        self.compile_assets()
-        base_egg_info.run(self)
-
+class assets_mixin:
     def compile_assets(self):
         try:
             subprocess.check_call(['npm', 'run', 'build'])
         except (OSError, subprocess.CalledProcessError) as e:
             print('Error compiling assets: ' + str(e))
             raise SystemExit(1)
+
+
+class sdist(base_sdist, assets_mixin):
+    def run(self):
+        self.compile_assets()
+        base_sdist.run(self)
+
+
+class bdist_egg(base_bdist_egg, assets_mixin):
+    def run(self):
+        self.compile_assets()
+        base_bdist_egg.run(self)
 
 
 setup(
@@ -105,5 +114,8 @@ setup(
         "wagtail>=1.4.0",
         "Django>=1.7.1",
     ],
-    cmdclass={'egg_info': egg_info},
+    cmdclass={
+        'sdist': sdist,
+        'bdist_egg': bdist_egg,
+    },
 )

